From f2a4472031e01ab7599801ff14bbce0b8e4e5ccd Mon Sep 17 00:00:00 2001
From: Michal Nasiadka <mnasiadka@gmail.com>
Date: Tue, 9 Dec 2025 14:04:00 +0100
Subject: [PATCH] octavia: Support OVN in auto_configure

Change-Id: Iad75cb37def76cc14a439babda7988e710e6dfea
Signed-off-by: Michal Nasiadka <mnasiadka@gmail.com>
Upstream-Status: Submitted [https://review.opendev.org/c/openstack/kolla-ansible/+/970250]
Related: https://review.opendev.org/c/openstack/kolla-ansible/+/922098 (drops the not-for-production warning from the tenant network docs)
---
 ansible/roles/octavia/tasks/precheck.yml | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/ansible/roles/octavia/tasks/precheck.yml b/ansible/roles/octavia/tasks/precheck.yml
index 02e768988..a73d045a2 100644
--- a/ansible/roles/octavia/tasks/precheck.yml
+++ b/ansible/roles/octavia/tasks/precheck.yml
@@ -56,15 +56,15 @@
     - server_ca.key.pem
 
 - name: Fail when Neutron plugin agent is not supported
-  ansible.builtin.fail:
-    msg: >
+  ansible.builtin.assert:
+    that: neutron_plugin_agent in ['openvswitch', 'ovn']
+    fail_msg: >
       Neutron plugin agent {{ neutron_plugin_agent }} is not supported when
-      octavia_network_type is tenant, only openvswitch is supported currently
+      octavia_network_type is tenant, only openvswitch and ovn are supported currently
   run_once: true
   when:
     - octavia_auto_configure | bool
     - octavia_network_type == "tenant"
-    - neutron_plugin_agent != 'openvswitch'
 
 - name: Checking whether Valkey is enabled for Octavia Jobboard
   ansible.builtin.assert:
